pageserver/src/tenant.rs

```diff
@@ -1503,9 +1510,18 @@ impl Tenant {
                                     "failed to activate timeline {}: {}",
                                     timeline.timeline_id, e
                                 ));
+
+                                timelines_broken_during_activation += 1;
                             }
                         }
                     }
+
+                    let elapsed = self.created_at.elapsed();
+                    let total_timelines = timelines_accessor.len();
+
+                    // log a lot of stuff, because some tenants sometimes suffer from user-visible
+                    // times to activate. see https://github.com/neondatabase/neon/issues/4025
+                    info!(since_creation_millis = elapsed.as_millis(), tenant_id = %self.tenant_id, activated_timelines, timelines_broken_during_activation, total_timelines, "Tenant activated");
```

Comment:
```suggestion
                    info!(since_creation_millis = elapsed.as_millis(), tenant_id = %self.tenant_id, activated_timelines, timelines_broken_during_activation, total_timelines, ?pre_state, ?post_state=?current_state, "finished");
```

The tenant might actually be broken at this point (See `broken_from_reason` call above).
The log message would be confusing in this case.
Suggestion: log pre and post state as fields, and change message to "done".

I'm ok with just having pre and post state as well.

Reply:
Aah, I missed the fact that tenant will be broken, possibly multiple times, fixing.

Reply:
Sadly the debug of TenantState includes the backtrace, which ... is unfortunate. I'll use the IntoStaticStr.

Reply:
https://gist.github.com/koivunej/2dbc16cb611f94341eaa7f59924a56a8 seems to work.

Reply:
fixed in 61ae7427.